docs_src/design/adr/0024-system-events.md

```diff
@@ -0,0 +1,93 @@
+# System Events ADR
+### Submitters
+- Lenny Goodell (Intel)
+
+## Change Log
+- [pending](https://github.com/edgexfoundry/edgex-docs/pull/795) (2022-07-12)
+
+## Referenced Use Case(s)
+- [System Events for Devices ](https://docs.edgexfoundry.org/2.3/design/ucr/0001-System-Events-for-Devices/)
+
+## Context
+System Events, aka Control Plane Events (CPE),  are new to EdgeX. This ADR addresses the [System Events for Devices](https://docs.edgexfoundry.org/2.3/design/ucr/0001-System-Events-for-Devices/) use case with an extensible design that can address other System Event use cases that may be identified in the future. This extensible design approach and the fact that System Events are produced and consumed by different EdgeX services makes it architecturally significant warranting this ADR.
+
+## Proposed Design
+To address the [System Events for Devices ](https://docs.edgexfoundry.org/2.3/design/ucr/0001-System-Events-for-Devices/) use case, Core Metadata will publish a new `SystemEvent` DTO to the EdgeX MessageBus when a device is added, updated or deleted. Consumers of these System Events will subscribe to the MessageBus to receive the new `SystemEvent` DTO .
+
+#### Data Transfer Object (DTO)
+
+This new `SystemEvent` DTO will contain the following data describing the System Event:
+
+- Event Type - i.e. device
+- Event Action - i.e. add/update/delete
+- Event Source - i.e core-metadata as the publisher
+- Event Owner - i.e device-onvif-camera as the device owner
+    - Optional based on the Event Type
+- Event Other- i.e. onvif-camera (profile name)
+    - Optional based on the Event Type
+- Event Details - i.e. Device DTO of newly added device or empty for Delete action.
+    - Optional and varies based on the Event Type and or Event Action. 
```

Comment:
We had better add a field whose type is an extendable map structure (`map[string]interface` in golang, for example).
It can be Event Other or Event Details, or we can add an Event Props or Event Attrs for this field.  All the naming is fine with me.

Reply:
The timestamp is also important here, so we need a field for that.

Reply:
Good catch on the `timestamp` . Will add.
I am trying to stay away (as much as possible) from implementation details such as data types (i.e. map[string]interface). That said I also struggled with Event Other. Maybe Event Tags, which infers a collection of Key/Value pairs?
For `Event Details` I think `varies based on` infers type is `any` (aka interface{}), but will add clarification.

Reply:
Added and reworked